### ISSUE TITLE
bug: syntax highlight broken

### ISSUE THREAD

**Issue by astorath** [hashicorp/vscode-terraform#160]
Syntax highlight is broken when hcl block starts on different line than block definition (it's a valid hcl syntax).

![image](https://user-images.githubusercontent.com/6577390/53625594-99de3580-3c14-11e9-80fc-d51f57a053e1.png)

![image](https://user-images.githubusercontent.com/6577390/53625601-9ea2e980-3c14-11e9-9972-e4be1e05b8b8.png)

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.

**Comment by radeksimko** [comment]
Hi @astorath 
I can see you opened the issue ~ 2 months before 0.12.0 was released, so this would have been a valid bug report at that point.

However such syntax is now considered invalid [since HCL2](https://github.com/hashicorp/hcl/blob/d7e7bca9a90f497e93c52038633506c398faec71/hclsyntax/parser.go#L316-L323), which was introduced into Terraform 0.12.

Any 0.12+ version will refuse to parse such config in plan, apply (or any other command which parses configuration) and provide the following error message:
```
Error: Invalid block definition

A block definition must have block content delimited by "{" and "}", starting
on the same line as the block header.
```

The extension will also highlight this fact if you have the language server enabled:

![Screenshot 2021-03-08 at 08 50 12](https://user-images.githubusercontent.com/287584/110297451-6694c500-7feb-11eb-81f6-cf976d128be0.png)


0.11 does treat it as valid and plan/apply will accept such config, but formatting such block of code via `terraform fmt` will provide a hint by putting the opening brace on the same line as the header.

```
$ cat main.tf
provider "aws"
{
  region = "eu-west-2"
}

$ cat main.tf | terraform fmt -
provider "aws" {
  region = "eu-west-2"
}
```

--- 

This VS Code extension (and the language server too) intend to support Terraform 0.12+ only - also due to the fact that the language (HCL) in these versions is more stabilized.

Therefore I believe this works as intended, so I'm going to close this issue.
